torch/_inductor/cudagraph_trees.py

```diff
@@ -250,7 +251,22 @@ def get_container(device_index: int):
         return container_dict[device_index]
 
 
-def cudagraphify_impl(
+def cudagraphify_impl(model, inputs, *args, **kwargs):
+    fn = None
+
+    def deferred_cudagraphify(second_inputs):
+        nonlocal fn
+        if fn is not None:
+            return fn(second_inputs)
+
+        assert inputs is second_inputs
+        fn, out = cudagraphify(model, inputs, *args, **kwargs)
```

Comment:
Uhhh don't you have to del the original arguments (model, inputs, args, kwargs)? Otherwise the deferred closure will hold onto them indefinitely??

Reply:
I can delete the deferred cudagraphify thing. It was just so that we do the actual running of the inputs when we initialize the CUDAGraphNode. But it's not needed if you think it makes it more confusing

Reply:
I'll instead change the cudagraphify so that the api for it is it returns the new compiled function and the new outputs.

Reply:
I think this is all wrapped up with the thing where you do a record/replay in the constructor and then do something special the first time run is called. I don't have a clear enough picture about how to make it clearer. If you can think of something that is easy to do and has more clarity, do it.